File: Scanner/CommunicationsFramework/Applications/Production/CNDService/Service/CNDServiceWinService/Service/CNDServerService.cs
using System.Collections.Generic;
using System;
using System.Diagnostics;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Linq;
using CommunicationsLibrary.CNDCommunicationsEnvironment.ComponentNetworkDirectoryService;
using UtilitiesLibrary.EventLoging;


namespace CNDServerService
{
	public partial class CNDServerWinService
	{
		
		//static void Main()
		//{
			//System.Windows.Forms.Application.Run(new CNDServerService());
			//}
			
			
#region  < DATA MEMBERS >
			
			private CNDService _CNDService;
			
#endregion
			
			
			protected override void OnStart(string[] args)
			{
				// Add code here to start your service. This method should set things
				// in motion so your service can do its work.
				try
				{
					this._CNDService = CNDService.GetInstance();
				}
				catch (Exception ex)
				{
					CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString());
				}
				
			}
			
			protected override void OnStop()
			{
				// Add code here to perform any tear-down necessary to stop your service.
				try
				{
					this._CNDService.Dispose();
				}
				catch (Exception ex)
				{
					CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString());
				}
			}
			
			protected override void OnShutdown()
			{
				try
				{
					base.OnShutdown();
				}
				catch (Exception ex)
				{
					CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString());
				}
				try
				{
					this._CNDService.Dispose();
				}
				catch (Exception ex)
				{
					CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString());
				}
			}
			
		}
		
	}

File: Scanner/CommunicationsFramework/Applications/Production/CNDService/Service/WindowsService/Service/CNDServerService.vb
Imports CommunicationsLibrary.CNDCommunicationsEnvironment.ComponentNetworkDirectoryService
Imports UtilitiesLibrary.EventLoging

Public Class CNDServerService


#Region " < DATA MEMBERS > "

    Private _CNDService As CNDService

#End Region


    Protected Overrides Sub OnStart(ByVal args() As String)
        ' Add code here to start your service. This method should set things
        ' in motion so your service can do its work.
        Try
            Me._CNDService = CNDService.GetInstance
        Catch ex As Exception
            CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString)
        End Try

    End Sub

    Protected Overrides Sub OnStop()
        ' Add code here to perform any tear-down necessary to stop your service.
        Try
            Me._CNDService.Dispose()
        Catch ex As Exception
            CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString)
        End Try
    End Sub

    Protected Overrides Sub OnShutdown()
        Try
            MyBase.OnShutdown()
        Catch ex As Exception
            CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString)
        End Try
        Try
            Me._CNDService.Dispose()
        Catch ex As Exception
            CustomEventLog.WriteEntry(EventLogEntryType.Error, ex.ToString)
        End Try
    End Sub

End Class
